glusterd2/main.go

```diff
@@ -119,6 +120,36 @@ func main() {
 		log.WithError(err).Fatal("Failed to generate local auth token")
 	}
 
+	// Get the Jaeger endpoints from the command line if provided
+	jaegerEndpoints := make(map[string]string)
+	jaegerEndpoint, _ := flag.CommandLine.GetString(tracing.JaegerEndpointOpt)
+	jaegerAgentEndpoint, _ := flag.CommandLine.GetString(tracing.JaegerAgentEndpointOpt)
+
+	// Create the Opencensus Jaeger exporter with info from command line
+	exporter := tracing.InitJaegerExporter(jaegerEndpoint, jaegerAgentEndpoint)
+	if exporter != nil {
+		defer exporter.Flush()
+		jaegerEndpoints[tracing.JaegerEndpointOpt] = jaegerEndpoint
+		jaegerEndpoints[tracing.JaegerAgentEndpointOpt] = jaegerAgentEndpoint
+		// Register the Jaeger exporter
+		tracing.RegisterJaegerExporter(exporter, jaegerEndpoints)
+	}
+
+	// Get the Jaeger endpoints from config file if provided
+	jaegerEndpoint2 := config.GetString(tracing.JaegerEndpointOpt)
+	jaegerAgentEndpoint2 := config.GetString(tracing.JaegerAgentEndpointOpt)
+	// Create the Opencensus Jaeger exporter with info from config file
+	if jaegerEndpoint != jaegerEndpoint2 || jaegerAgentEndpoint != jaegerAgentEndpoint2 {
+		exporter := tracing.InitJaegerExporter(jaegerEndpoint2, jaegerAgentEndpoint2)
```

Comment:
You could actually do all this within the tracing package. The `InitJaegerExporter` function can fetch the configured options and if required create the exporter and register the endpoints. It would then return just the exporter (or nil) so it could be closed when GD2 exits.

This way much of the control over tracing initialization is within the tracing package, and the main package and function remain simple.